packages/editor/src/components/browser/function/FunctionBrowser.tsx

```diff
@@ -166,16 +161,24 @@ const FunctionBrowser = (props: { value: string; onChange: (value: BrowserValue)
         ref={parentRef}
       >
         <TableBody>
-          {virtualizer.getVirtualItems().map(virtualRow => {
-            const row = rows[virtualRow.index];
-            return (
-              <SelectRow key={row.id} row={row} onDoubleClick={props.onDoubleClick}>
-                {row.getVisibleCells().map(cell => (
-                  <TableCell key={cell.id}>{flexRender(cell.column.columnDef.cell, cell.getContext())}</TableCell>
-                ))}
-              </SelectRow>
-            );
-          })}
+          {rows.some(row => row.depth === 1 && row.getIsExpanded())
+            ? virtualizer.getVirtualItems().map(virtualRow => {
+                const row = rows[virtualRow.index];
+                return (
+                  <SelectRow key={row.id} row={row} onDoubleClick={props.onDoubleClick}>
+                    {row.getVisibleCells().map(cell => (
+                      <TableCell key={cell.id}>{flexRender(cell.column.columnDef.cell, cell.getContext())}</TableCell>
+                    ))}
+                  </SelectRow>
```

Comment:
duplicated now. maybe strip out shared component for this?

Reply:
Sure, I can do that. Should I implement this new Row component at the browser level, as I could reuse it for the Role, Type, and Attribute Browsers as well? However, this will likely become unnecessary soon, as we plan to use the UI component browser soon.
![grafik](https://github.com/user-attachments/assets/070ac7d4-d522-4a5a-97b3-1b0ffc5e274e)

Reply:
why not 👍